include/boost/geometry/algorithms/detail/relate/boundary_checker.hpp

```diff
@@ -94,42 +98,33 @@ class boundary_checker<Geometry, Strategy, multi_linestring_tag>
 
     // First call O(NlogN)
     // Each next call O(logN)
-    template <boundary_query BoundaryQuery>
-    bool is_endpoint_boundary(point_type const& pt) const
+    template <boundary_query BoundaryQuery, typename Point>
+    bool is_endpoint_boundary(Point const& pt) const
     {
-        typedef geometry::less<point_type, -1, typename Strategy::cs_tag> less_type;
+        using less_type = geometry::less<mutable_point_type, -1, typename Strategy::cs_tag>;
 
-        typedef typename boost::range_size<Geometry>::type size_type;
-        size_type multi_count = boost::size(m_geometry);
+        auto const multi_count = boost::size(m_geometry);
 
         if ( multi_count < 1 )
+        {
             return false;
+        }
 
         if ( ! m_is_filled )
         {
             //boundary_points.clear();
             m_boundary_points.reserve(multi_count * 2);
 
-            typedef typename boost::range_iterator<Geometry const>::type multi_iterator;
-            for ( multi_iterator it = boost::begin(m_geometry) ;
-                  it != boost::end(m_geometry) ; ++ it )
+            for (auto const& ls : m_geometry)
```

Comment:
This is incorrect. MultiLinestring has to be Boost.Range not STD range.

In general we cannot use range based for loop for a geometry that can be passed by the user, only for geometries and containers that we create ourselves internally.

Reply:
All right I will revert this. Thanks.

Reply:
:heavy_check_mark: